cryptography/src/darwinMain/kotlin/com/wire/kalium/cryptography/utils/CryptoUtils.kt

```diff
@@ -1,80 +1,145 @@
+import platform.CoreCrypto.kCCOptionPKCS7Padding
+import platform.CoreCrypto.kCCSuccess
+import platform.Security.SecRandomCopyBytes
+import platform.Security.errSecSuccess
+import platform.Security.kSecRandomDefault
 
-actual fun calcMd5(bytes: ByteArray): String {
-    val digestData = UByteArray(CC_MD5_DIGEST_LENGTH)
-    val data = toData(bytes)
-    CC_MD5(data.bytes, data.length.toUInt(), digestData.refTo(0))
-
-    return toData(digestData.asByteArray()).base64Encoding()
+actual fun encryptDataWithAES256(data: PlainData, key: AES256Key): EncryptedData {
+    val outputBuffer = Buffer()
+    encryptFileWithAES256(Buffer().write(data.data), key, outputBuffer)
+    return EncryptedData(outputBuffer.readByteArray())
 }
 
-actual fun calcSHA256(bytes: ByteArray): ByteArray {
-    TODO("Not yet implemented")
+actual fun decryptDataWithAES256(data: EncryptedData, secretKey: AES256Key): PlainData {
+    val outputBuffer = Buffer()
+    decryptFileWithAES256(Buffer().write(data.data), outputBuffer, secretKey)
+    return PlainData(outputBuffer.readByteArray())
 }
 
-@Suppress("TooGenericExceptionCaught")
-actual fun calcFileMd5(dataSource: Source): String? =
+actual fun encryptFileWithAES256(assetDataSource: Source, key: AES256Key, outputSink: Sink): Long {
     try {
-        dataSource.buffer().use { source ->
-            HashingSink.md5(blackholeSink()).use { sink ->
-                source.readAll(sink)
-                sink.hash.toByteArray().encodeBase64()
+        val iv = generateRandomData(kCCBlockSizeAES128.toInt())
+        val plainData = assetDataSource.buffer().readByteArray()
+        val encryptedBuffer = ByteArray(plainData.size + kCCBlockSizeAES128.toInt())
+
+        // TODO avoid read whole file into memory by using streaming or block based API
+        return memScoped {
+            val bytesCopied = alloc<ULongVar>()
+            val status = key.data.usePinned { key ->
+                iv.usePinned { iv ->
+                    plainData.usePinned { plainData ->
+                        encryptedBuffer.usePinned { encryptedBuffer ->
+                            CCCrypt(
+                                kCCEncrypt,
+                                kCCAlgorithmAES,
+                                kCCOptionPKCS7Padding,
+                                key.addressOf(0),
+                                kCCKeySizeAES256.toULong(),
+                                iv.addressOf(0),
+                                plainData.addressOf(0),
+                                plainData.get().size.toULong(),
+                                encryptedBuffer.addressOf(0),
+                                encryptedBuffer.get().size.toULong(),
+                                bytesCopied.ptr
+                            )
+                        }
+                    }
+                }
+            }
+
+            if (status != kCCSuccess) {
+                throw Error("Failure while encrypting data using AES256")
```

Comment:
Detekt says this error is too generic, so should I create a custom error or is there standard error which makes sense is in this case?